pytest_asyncio_network_simulator/plugin.py

```diff
@@ -0,0 +1,35 @@
+import asyncio
+from typing import (
+    Any,
+    AsyncGenerator,
+)
+
+import pytest
+
+from .network import (
+    Network,
+)
+from .router import (
+    Router,
+)
+
+
+@pytest.fixture
+async def router() -> AsyncGenerator[Router, None]:
```

Comment:
Something feels awkward about defining the fixture here. I suppose the project that wants to use it would importing `router` directly into its conftest. (My issue is the surprise of adding a fixture just by doing an import)

Maybe `Router()` could become a context manager, and then any project that wants to use it can add a fixture in their own conftest, like:

```py
from pytest_asyncio_network import Router

@pytest.fixture
def router():
  with Router() as running_router:
    yield running_router
```

Without *too* much boilerplate.

Reply:
Ah interesting, TIL that this is how you load a pytest plugin:
```
'pytest11': ['pytest-asyncio-network-simulator=pytest_asyncio_network_simulator.plugin'],
```

---

> What do you think of the following

Yeah, both of those examples look great. (with preference for the context manager, for people who will use the first example they find :))

Reply:
see: https://github.com/ethereum/pytest-asyncio-network-simulator/pull/4